### PR TITLE
Optim convlstm

### PR DESCRIPTION
**Pull request by dding3**
## What changes were proposed in this pull request?
1. Add prepotopy for convlstm
2. Add convlstm doc
3. Some minor change

## How was this patch tested?
ut

### CODE REVIEW

**Comment by JerryYanWan** [comment]
LGTM, please test on jenkins :)

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala [hunk 1]
```diff
@@ -43,35 +43,37 @@ import scala.reflect.ClassTag
...
-    val i2g = Sequential()
-      .add(Contiguous())
-      .add(SpatialConvolution(inputSize, hiddenSize, kernelI, kernelI,
-      stride, stride, kernelI/2, kernelI/2, wRegularizer = wRegularizer,
-        bRegularizer = bRegularizer))
+  override def preTopology: AbstractModule[Activity, Activity, T] =
+    Sequential()
+        .add(TimeDistributed(SpatialConvolution(inputSize, outputSize*4, kernelI, kernelI,
+          stride, stride, kernelI/2, kernelI/2, wRegularizer = wRegularizer,
+          bRegularizer = bRegularizer)))
```
Comment (JerryYanWan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala]: I add a   `override def getPreTopologyNOutput: Int` in Cell. It is used to insert a BatchNormalization layer between preTopology and topology. If preTopology returns a Sequential, please all insert this line:

`  override def getPreTopologyNOutput: Int = 4 * outputSize`

Comment (JerryYanWan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala]: A model definitin can be:

To insert a BatchNormalization layer:
```
val model = Sequential[Double]()
      .add(Recurrent[Double](BatchNormParams())
        .add(LSTMPeephole[Double](inputSize, hiddenSize)))
```

Or Not:
```

val model = Sequential[Double]()
      .add(Recurrent[Double]()
        .add(LSTMPeephole[Double](inputSize, hiddenSize)))
```

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala [hunk 2]
```diff
@@ -43,35 +43,37 @@ import scala.reflect.ClassTag
...
```
Comment (dding3) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala]: In convlstm, there is no dropout.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala (hunk 1 above)
Comment (dding3) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala]: For spatialconvolution, batchnormalization is a little different. Let me do it in another pr.
